packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -202,70 +184,103 @@ function toTimestamp(date: number | Date) {
   return date;
 }
 
-function isDailyTriggerInput(
-  trigger: SchedulableNotificationTriggerInput
-): trigger is DailyTriggerInput {
-  if (typeof trigger !== 'object') return false;
-  const { channelId, ...triggerWithoutChannelId } = trigger as DailyTriggerInput;
-  return (
-    Object.keys(triggerWithoutChannelId).length ===
-      DAILY_TRIGGER_EXPECTED_DATE_COMPONENTS.length + 1 &&
-    DAILY_TRIGGER_EXPECTED_DATE_COMPONENTS.every(
-      (component) => component in triggerWithoutChannelId
-    ) &&
-    'repeats' in triggerWithoutChannelId &&
-    triggerWithoutChannelId.repeats === true
-  );
+function parseDailyTrigger(trigger: NotificationTriggerInput): NativeDailyTriggerInput | undefined {
+  if (
+    trigger !== null &&
+    typeof trigger === 'object' &&
+    'type' in trigger &&
+    trigger.type === CalendarTriggerTypes.DAILY
+  ) {
+    validateDateComponentsInTrigger(trigger, ['hour', 'minute']);
+    const result: NativeDailyTriggerInput = {
+      type: 'daily',
+      hour: trigger.hour ?? placeholderDateComponentValue,
+      minute: trigger.minute ?? placeholderDateComponentValue,
+    };
+    if (trigger.channelId) {
+      result.channelId = trigger.channelId;
+    }
+    return result;
+  }
+  return undefined;
 }
 
-function isWeeklyTriggerInput(
-  trigger: SchedulableNotificationTriggerInput
-): trigger is WeeklyTriggerInput {
-  if (typeof trigger !== 'object') return false;
-  const { channelId, ...triggerWithoutChannelId } = trigger as WeeklyTriggerInput;
-  return (
-    Object.keys(triggerWithoutChannelId).length ===
-      WEEKLY_TRIGGER_EXPECTED_DATE_COMPONENTS.length + 1 &&
-    WEEKLY_TRIGGER_EXPECTED_DATE_COMPONENTS.every(
-      (component) => component in triggerWithoutChannelId
-    ) &&
-    'repeats' in triggerWithoutChannelId &&
-    triggerWithoutChannelId.repeats === true
-  );
+function parseWeeklyTrigger(
+  trigger: NotificationTriggerInput
+): NativeWeeklyTriggerInput | undefined {
+  if (
+    trigger !== null &&
+    typeof trigger === 'object' &&
+    'type' in trigger &&
+    trigger.type === CalendarTriggerTypes.WEEKLY
+  ) {
+    validateDateComponentsInTrigger(trigger, ['weekday', 'hour', 'minute']);
+    const result: NativeWeeklyTriggerInput = {
+      type: 'weekly',
+      weekday: trigger.weekday ?? placeholderDateComponentValue,
+      hour: trigger.hour ?? placeholderDateComponentValue,
+      minute: trigger.minute ?? placeholderDateComponentValue,
+    };
+    if (trigger.channelId) {
+      result.channelId = trigger.channelId;
```

Comment:
not really important: it seems the `channelId` doesn't need to be added conditionally, it can be present always (and potentially be undefined). I find it's a little simpler with fewer conditions